extras/indexing/src/main/java/org/apache/rya/indexing/entity/storage/mongo/MongoEntityStorage.java

```diff
@@ -242,4 +281,46 @@ private static Bson makeExplicitTypeFilter(final RyaURI typeId) {
 
         return Stream.of(dataTypeFilter, valueFilter);
     }
+
+    private boolean detectDuplicates(final Entity entity) throws EntityStorageException {
+        boolean hasDuplicate = false;
+        if (duplicateDataDetector.isDetectionEnabled()) {
+            if (mongoTypeStorage == null) {
+                mongoTypeStorage = new MongoTypeStorage(mongo, ryaInstanceName);
+            }
+            final Builder builder = new Builder();
+            builder.setSubject(entity.getSubject());
+            boolean abort = false;
+            for (final RyaURI typeRyaUri : entity.getExplicitTypeIds()) {
+                Optional<Type> type;
+                try {
+                    type = mongoTypeStorage.get(typeRyaUri);
+                } catch (final TypeStorageException e) {
+                    throw new EntityStorageException("Unable to get entity type: " + typeRyaUri, e);
+                }
+                if (type.isPresent()) {
+                    final ConvertingCursor<TypedEntity> cursor = search(Optional.empty(), type.get(), Collections.emptySet());
+                    while (cursor.hasNext()) {
```

Comment:
Oops, it's only grabbing one Entity to compare.  I reworked so it now finds a set of potential Entities to compare based on them having all the same explicit type IDs.

The subjects don't matter and querying for properties doesn't help us since we're trying to find properties that are CLOSE but not quite equal.  That leaves us with only the Types to narrow our initial search of Entities to check. Once we grab the Entities the (near) duplicate data detector is run over them.